pkg/checks/external/nodeCheck/main.go

```diff
@@ -0,0 +1,99 @@
+package nodeCheck
+
+import (
+	"errors"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/api/v1/pod"
+)
+
+// checkNode checks the node's age to make sure it's not less than three minutes old. If so, sleep for one minute
+// and check if kube proxy is ready and running on the node before running the check. This function requires the
+// khcheck to expose the pod's node name information using environment variables.
+func CheckNode(client *kubernetes.Clientset, nodeName string, now time.Time) {
```

Comment:
How about the caller passes in the pod name (and maybe namespace) instead of the node name?  What if we just go one step further and assume the pod's hostname is the result of `os.Hostname()`?

Also, can we rename `CheckNode` here to something more descript?  The current call is `nodeCheck.CheckNode()` which is very generic.  Perhaps rename to `nodeCheck.WaitForKubeProxy`?  I know that we have a `waitForKubeProxyReady` func, but that isn't exposed to the user for calling.